functions/floris_compute_windspeed.m

```diff
@@ -1,20 +1,28 @@
 function [ dwTurbs ] = floris_compute_windspeed( turbines,wakes,inputData,wt_rows,turbirow )
 
-    for  dw_turbi = wt_rows{turbirow+1}% for all turbines in dw row
-        % Sum of kinetic energy deficits
-        sumKed  = 0; % outer sum of Eq. 22
-        % Turbulence intensity vector
-        TiVec = inputData.TI_0;
-        for uw_turbrow = 1:turbirow % for all rows upstream of this current row
+
+    % Calculate the effect of all upstream turbines on the current turbine row
+    for  dw_turbi = wt_rows{turbirow+1}
+        % Calculate the effect of all upstream turbines on the current turbine
+        sumKed  = 0; % Sum of kinetic energy deficits (outer sum of Eq. 22)
+        TiVec = inputData.TI_0; % Turbulence intensity vector
+        
+        for uw_turbrow = 1:turbirow % for all turbine rows upstream of this current turbine
             for uw_turbi = wt_rows{uw_turbrow} % for each turbine in that row
+                % displacement in x-direction between uw_turbi and dw_turbi [m]
                 deltax = turbines(dw_turbi).LocWF(1)-turbines(uw_turbi).LocWF(1);
                 [~,turbLocIndex] = min(abs(wakes(uw_turbi).centerLine(1,:)-turbines(dw_turbi).LocWF(1)));
                 
                 % Q is the normalized velocity deficit on the turbine swept area
                 if strcmp(inputData.wakeType,'Zones') && strcmp(inputData.atmoType,'uniform')
-                    Q = FlorisQ();
+                    Q = FlorisQ(); % Herein wecalculate overlap areas between 
+                    % different wake zones of uw_turbi and the rotor plane 
+                    % of dw_turbi.
                 else
-                    Q = integralQ();
+                    Q = integralQ(); % Herein we use a numerical integral 
+                    % function to  determine the overlap ratios between a 
+                    % non-circular wake shape (e.g. gaussian) and the rotor 
```

Comment:
The wake shape doesn't have to be non-circular, it can just be any shape really. Another important note is that this works best on velocity profiles that are continuous. That is actually why the original zoned model has it's own function, FlorisQ(). Because it was available, fast and the numerical integration method doesn't work very well for the discrete jumps in velocity.